hazelcast/src/main/java/com/hazelcast/sql/impl/extract/AbstractGenericExtractor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.extract;
+
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+/**
+ * Base class for generic extractors.
```

Comment:
What does "generic" means in extractors context? Looks like all extractors are generic now.

Reply:
The idea is that is uses `Extractors`, capable of extracting data from any underlying object. In the prototype branches (`sql` and `sql-compiler`) we have other extractor implementations that either use reflection or cached `PortableReader`. It is not very clear how we are going to organize it in the end, until we try to cover different edge cases for Java classes, portables, and perhaps the new serialization formats. This is a separate task.